django_eth_events/chainevents.py

```diff
@@ -1,25 +1,41 @@
 from abc import ABCMeta, abstractmethod
+from typing import Optional, List
 
 from .singleton import SingletonABCMeta
 
 
-class AbstractAddressesGetter(object):
+class AbstractAddressesGetter(metaclass=SingletonABCMeta):
     """Abstract AddressesGetter class."""
-    __metaclass__ = SingletonABCMeta
 
     @abstractmethod
-    def get_addresses(self): pass
+    def get_addresses(self) -> List: pass
 
     @abstractmethod
-    def __contains__(self, address): pass
+    def __contains__(self, address: str) -> bool: pass
 
 
-class AbstractEventReceiver(object):
+class AbstractEventReceiver(metaclass=ABCMeta):
     """Abstract EventReceiver class."""
-    __metaclass__ = ABCMeta
 
     @abstractmethod
-    def save(self, decoded_event, block_info): pass
+    def save(self, decoded_event: dict, block_info: dict) -> Optional[object]:
```

Comment:
For typing, you should use Dict instead of dict. (`from typing import Dict`) Then you can do `Dict[str, any]` for a dictionary where the keys are strings and the values can have different types. If you now the types you can even do something like `Dict[str, Union[int, str, bytes]]`, but I think `Dict[str, any]` is enough :)

Reply:
Great, makes sense to use Dict type instead of primitive dict.

Reply:
Changes applied, thanks @Uxio0 !